### PR TITLE
feat: copilot plugin questions

### PR DESCRIPTION
**Pull request by yuqizhou77**
- add excludesEnum for string validation to validate that a string does not equal to any items in the list
- add Copilot_Plugin feature flag
- add scaffolding questions and options for copilot plugin
- returns mock data for now when need to validate api spec, which will integrate with api spec parser later when all components are ready.
- screeshots:
image

After selecting the first item
image

After selecting to start with an OpenAPI specification
image

After selecting to start from OpenAI plugin
image

vscode-extension:
- selectFileorInput: fix the issue for "back" in input box so that it can back to the question to select between "browse" and "input"

### CODE REVIEW

**Review by jayzhang — DISMISSED** [review]

**Review by tecton — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -252,7 +252,25 @@
...
   "core.createProjectQuestion.projectType.tab.detail": "Embed your own web content in Teams, Outlook, and the Micosoft 365 app",
   "core.createProjectQuestion.projectType.tab.title": "App Features Using a Tab",
+  "core.createProjectQuestion.projectType.copilotPlugin.detail": "Create a plugin to extend Copilot using your APIs",
+  "core.createProjectQuestion.projectType.copilotPlugin.label": "Plugin for Copilot",
+  "core.createProjectQuestion.projectType.copilotPlugin.title": "Plugin for Copilot",
+  "core.createProjectQuestion.projectType.copilotPlugin.placeholder": "Select an option",
   "core.createProjectQuestion.title": "New Project",
+  "core.createProjectQuestion.capability.copilotPluginNewApiOption.label": "Start with a new API",
+  "core.createProjectQuestion.capability.copilotPluginNewApiOption.detail": "Create a plugin with a new API from Azure Functions",
+  "core.createProjectQuestion.capability.copilotPluginApiSpecOption.label": "Start with an Open API Specification",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:262]: OpenAPI is a single word.

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -252,7 +252,25 @@
...
+  "core.createProjectQuestion.capability.copilotPluginApiSpecOption.detail": "Create a plugin from your existing API",
+  "core.createProjectQuestion.capability.copilotPluginAIPluginOption.label": "Start with an Open AI Plugin",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:264]: Open AI -> OpenAI

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -252,7 +252,25 @@
...
+  "core.createProjectQuestion.capability.copilotPluginAIPluginOption.detail": "Convert an Open AI Plugin to Microsoft 365 Copilot plugin",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:265]: OpenAI

File: packages/fx-core/resource/package.nls.json [hunk 4]
```diff
@@ -252,7 +252,25 @@
...
+  "core.createProjectQuestion.apiSpec.title": "Open API Spec",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:266]: OpenAPI

File: packages/fx-core/resource/package.nls.json [hunk 5]
```diff
@@ -252,7 +252,25 @@
...
+  "core.createProjectQuestion.apiSpec.placeholder": "Enter Open API Specification URL",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:267]: OpenAPI

File: packages/fx-core/resource/package.nls.json [hunk 6]
```diff
@@ -252,7 +252,25 @@
...
+  "core.createProjectQuestion.apiSpecInputUrl.label": "Enter Open API Specification Location",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:268]: OpenAPI

File: packages/fx-core/resource/package.nls.json [hunk 7]
```diff
@@ -252,7 +252,25 @@
...
+  "core.createProjectQuestion.AIPluginManifest.title": "Open AI Plugin Manifest",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:269]: OpenAI

File: packages/fx-core/resource/package.nls.json [hunk 8]
```diff
@@ -252,7 +252,25 @@
...
+  "core.createProjectQuestion.AIPluginManifest.placeholder": "Enter Open AI Plugin Manifest URL",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:270]: OpenAI

File: packages/fx-core/resource/package.nls.json (hunk 2 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json]: Thanks! updated.

File: packages/fx-core/resource/package.nls.json (hunk 4 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json]: updated.

File: packages/fx-core/resource/package.nls.json (hunk 6 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json]: updated.

File: packages/fx-core/resource/package.nls.json (hunk 7 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json]: updated.

File: packages/fx-core/resource/package.nls.json (hunk 8 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json]: updated.

File: packages/fx-core/resource/package.nls.json (hunk 5 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json]: updated.

**Review by sffamily — DISMISSED** [review]

**Review by nliu-ms — DISMISSED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]

**Review by tecton — APPROVED** [review]

**Review by jayzhang — DISMISSED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by sffamily — APPROVED** [review]

**Pull request merged**